api/controllers/MerchStoreController.ts

```diff
@@ -181,15 +187,32 @@ export class MerchStoreController {
     return { error: null, order };
   }
 
-  @Patch('/order')
-  async editMerchOrder(@Body() fulfillOrderRequest: FulfillMerchOrderRequest,
+  @Patch('/order/:uuid')
+  async editMerchOrder(@Params() params: UuidParam,
+    @Body() editOrderRequest: EditMerchOrderRequest,
     @AuthenticatedUser() user: UserModel): Promise<EditMerchOrderResponse> {
+    if (!PermissionsService.canAccessMerchStore(user)) throw new ForbiddenError();
+    const { status, pickupEvent } = editOrderRequest.order;
+    // status and pickup event are not allowed to be updated at the same time,
+    // since they are not logically related, and thus should never have to be updated at the same request
+    if (status && pickupEvent) throw new UserError('Only status or pickupEvent can be updated at once, not both');
+    // members are only allowed to cancel orders. store admins can perform any operation
```

Comment:
> Can an order be canceled and then re-placed? Should that be allowed? If it is, then we need to update quantities and all that.

No, this isn't allowed per store requirements.

> Should this endpoint be split into separate member and admin edits? e.g. DELETE /order/:uuid for members canceling orders and PATCH /order/:uuid for admins canceling orders or something?

I like the idea of splitting the giant `PATCH /order/:uuid` route into separate routes like `/cancellation` or `/missed` or `/fulfillment`. I think in the case with member vs. admin edit, it should be fine to keep them in the same route, since we do the same sort of split with other routes. Also I just realized I wrote some buggy ass code where if an admin cancells a user's order, then the admin gets refunded LMAO